### PR TITLE
timer: implement table based `TimerStore`

### LINKED ISSUE #44445: Implement table based `TimerStore`

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #44445

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: timer/tablestore/sql.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/timer/api"
+)
+
+// CreateTimerTableSQL returns a SQL to create timer table
+func CreateTimerTableSQL(dbName, tableName string) string {
```
Comment (YangKeao) [inline timer/tablestore/sql.go:26]: This table is not added to the `bootstrap` stage in this PR, right?

**Review by YangKeao — APPROVED** [review]
/lgtm

File: timer/tablestore/sql.go (hunk 1 above)
Comment (lcwangchao) [reply timer/tablestore/sql.go:26]: yes, it will be added to bootstrap if some feature want to use it

**Review by hawkingrei — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Pull request merged**